config/default.yml

```diff
@@ -1658,6 +1658,15 @@ Rails/Blank:
   # Convert usages of `unless present?` to `if blank?`
   UnlessPresent: true
 
+Rails/BulkChangeTable:
+  Database: null
+  SupportedDatabases:
+    - mysql
+    - postgresql
+    - '...'
```

Comment:
I'm sorry if my understanding was wrong.

Is this to change this default.yml to match the ConnectionAdapter of Rails application that user is targeting?

I tried this cop. I ran the ConnectionAdapter as "Oracle", but false positive will occur if I run with this default setting. Perhaps the same is true when using "SQLite3" (Other than "MySQL", "PostgreSQL").

If it is need to change depending on the ConnectionAdapter users are using, it seems not to be the default.

The ConnectionAdapter used by Rails application can be determined by `adapter` in Rails.root/config/database.yml. The core adapters managed by Rails is as follows.

- MySQL => mysql2
- PostgreSQL => postgresql
- SQLite3 => sqlite3

Rather than preparing the default.yml setting, one option is to automatically decide based on `adapter` specified in Rails.root/config/database.yml.

However, this proposal is not silver bullet, connection to multiple databases using `establish_connection` etc seems difficult to detect by static analysis.

In any case, it seems that it is not good for false positives to appear depending on the dependent ConnectionAdapter.

Reply:
Thanks for your reviewing!

> Is this to change this default.yml to match the connection adapter of Rails application that user is targeting?

Yes. For example, if using MySQL, it is needed to edit `.rubocop.yml` as follows:

```yaml
Rails/BulkChangeTable:
  Database: mysql
```

For that reason, I think it will occur false positives on the other adapters by default.
I think there are two options to this Cop.

- If the adapter is not set, this Cop will not detect offenses
- Automatically determine adapter from `database.yml`

The former will be occur false negatives on the MySQL adapter or the PostgreSQL adapter. And also, it will also behave the same as disabled by default.
The latter looks good, but as @koic says, there are some difficulties. For example, how can this Cop determine `Rails.root`? What options should be provided to users using multiple databases?

Currently I'm going to fix this Cop based on the latter, but I'd like to hear many opinions.

Reply:
> For example, how can this Cop determine Rails.root?

I think that the root of repository can be regarded as `Rails.root`. Of course, if it can be changed in configuration, then it can correspond to the unexpected configuration, but first I think that it is better to start with support of CoC configuration that ride on Rails way (I think it is better to start with the smallest case) .

> What options should be provided to users using multiple databases?

Since the regular configuration of Rails (5.2 or lower) doesn't yet support multiple databases, I think that this cop don't need to support yet.

📝 Rails 6 introduces Part 1 implementation of multiple databases.
rails/rails#32274